src/eval/fixpoint.rs

```diff
@@ -203,3 +203,43 @@ pub fn patch_field<C: Cache>(
 
     Ok(())
 }
+
+/// Revert an evaluated record (`Record`) back to a recursive record (`RecRecord`). The fixpoint
+/// will be recomputed again at evaluation.
+///
+/// `revert` is used when an operation (excluding `merge`, which has its own reverting logic) might
+/// have changed the value of fields (e.g. a lazy contract application on a record), requiring to
+/// update recursive dependencies.
+///
+/// # Parameters
+///
+/// `env` and `local_env` are similar to the parameters of
+/// `[crate::transform::Closurize::closurize]`.
+///
+/// - `cache`: the evaluation cache
+/// - `record_data`: the data of the record to revert
+/// - `env`: the final environment in which the fields of the result will be closurized
+/// - `local_env`: the environment of the record represented by `record_data`
+pub fn revert<C: Cache>(
+    cache: &mut C,
+    record_data: RecordData,
+    env: &mut Environment,
+    local_env: &Environment,
+) -> Term {
+    let fields = record_data
+        .fields
+        .into_iter()
+        // In fact, we don't really need to closurize, and could revert alone. However, that would
+        // mean duplicating code from
```

Comment:
Oh, I thought we could be slightly more efficient by doing stuff "in-place", but in fact I realized midway that this is false: we need to closurize - or do something similar - anyway, because reverting allocates a new thunk, and environment is immutable (well, we can shadow, but at this point this isn't tremendously useful).

I'll just get rid of this comment altogether.